src/esp/physics/ArticulatedObject.h

```diff
@@ -553,6 +670,102 @@ class ArticulatedObject : public esp::physics::PhysicsObjectBase {
     Magnum::Debug{} << "No base implementation of \"clampJointLimits\". ";
   }
 
+  //=========== Joint Motor API ===========
+
+  /**
+   * @brief Create a new JointMotor from a JointMotorSettings.
+   *
+   * Note: No base implementation. See @ref bullet::BulletArticulatedObject.
+   * @param index DoF (for revolute or prismatic joints) or Link (spherical
+   * joints)
+   * @param settings The settings for the joint motor. Must have JointMotorType
+   * correctly configured.
+   * @return The motorId for the new joint motor or ID_UNDEFINED (-1) if failed.
+   */
+  virtual int createJointMotor(
+      CORRADE_UNUSED const int index,
+      CORRADE_UNUSED const JointMotorSettings& settings) {
+    Magnum::Debug{} << "No base implementation of \"createJointMotor\". "
+                       "Requires a physics simulator implementation.";
+    return ID_UNDEFINED;
+  }
+
+  /**
+   * @brief Remove and destroy a joint motor.
+   */
+  virtual void removeJointMotor(const int motorId) {
+    CHECK(jointMotors_.count(motorId) > 0);
+    jointMotors_.erase(motorId);
+  }
+
+  /**
+   * @brief Get a copy of the JointMotorSettings for an existing motor.
+   */
+  virtual JointMotorSettings getJointMotorSettings(const int motorId) {
+    CHECK(jointMotors_.count(motorId) > 0);
+    return jointMotors_.at(motorId)->settings;
+  }
+
+  /**
+   * @brief Update a JointMotor with new settings.
+   */
+  virtual void updateJointMotor(const int motorId,
+                                const JointMotorSettings& settings) {
+    CHECK(jointMotors_.count(motorId) > 0);
+    CHECK(jointMotors_.at(motorId)->settings.motorType == settings.motorType);
+    jointMotors_.at(motorId)->settings = settings;
+  }
+
+  /**
+   * @brief Query a map of motorIds -> links/joints for all active JointMotors.
+   */
+  virtual std::map<int, int> getExistingJointMotors() {
+    std::map<int, int> motorIdsToLinkIds;
```

Comment:
Reserve size of map

Reply:
std::map has no reserve without custom allocator.

Reply:
Actually, wait jointMotors is already a std::map right? just use the copy constructor.

Reply:
@aclegg3 Went ahead and did it with an iterator constructor which should be faster.